### PR TITLE
add can init

### PR DESCRIPTION
**Pull request by pavelskipenes**

### CODE REVIEW

**Review by evenkjaexli — APPROVED** [review]

**Comment by pavelskipenes** [comment]
Unidirectional CAN bus setup. Node 1 is producer, Node 2 is consumer. `LOOPBACK` mode never worked. Interrupts never worked on node 1.

**Comment by pavelskipenes** [comment]
closes #14 #22

File: nodes/node_1/src/modules/include/mcp2515.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+void mcp2515_can_tx(uint16_t id, union can_data data);
+void mcp2515_set_mode(enum can_mode);
+
+// reception and interrupts are not supported
+#if 0
+void mcp2515_can_rx(can_frame *);
+void handle_interrupt();
+enum mcp2515_interrupt_source
+{
+	CMP2515_INT_RX0_FULL,
```
Comment (evenkjaexli) [inline nodes/node_1/src/modules/include/mcp2515.h:20]: MCP2515

File: nodes/node_1/src/modules/include/mcp2515.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	CMP2515_INT_RX1_FULL,
+	CMP2515_INT_TX0_EMPTY,
+	CMP2515_INT_TX1_EMPTY,
+	CMP2515_INT_TX2_EMPTY,
+	CMP2515_INT_ERROR,
+	CMP2515_INT_WAKE_UP,
+	CMP2515_INT_MESSAGE_ERROR,
```
Comment (evenkjaexli) [inline nodes/node_1/src/modules/include/mcp2515.h:27]:
```suggestion
	MCP2515_INT_RX0_FULL,
	MCP2515_INT_RX1_FULL,
	MCP2515_INT_TX0_EMPTY,
	MCP2515_INT_TX1_EMPTY,
	MCP2515_INT_TX2_EMPTY,
	MCP2515_INT_ERROR,
	MCP2515_INT_WAKE_UP,
	MCP2515_INT_MESSAGE_ERROR,
```

File: nodes/node_1/src/modules/mcp2515.c [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+ * @param mcp2515 pointer to gpio slave
+ * @param enum mcp2515_instruction_t to run
+ */
+static void mcp2515_instruction(GPIO *p_mcp2515, enum mcp2515_instruction_t instruction)
+{
+	// instructions are only valid if they follow a falling CS#
+	// keep the signal low. This works as long as there are only one SPI device connected.
+	gpio_set(p_mcp2515, GPIO_HIGH);
+	gpio_set(p_mcp2515, GPIO_LOW);
+	spi_send_byte(p_mcp2515, instruction);
```
Comment (evenkjaexli) [inline nodes/node_1/src/modules/mcp2515.c:169]:
```suggestion
	gpio_set(p_mcp2515, GPIO_HIGH);
	spi_send_byte(p_mcp2515, instruction);
	gpio_set(p_mcp2515, GPIO_LOW);
```

**Review by evenkjaexli — COMMENTED** [review]
fix enum mcp2515_interrupt_source in mcp2515.h
and change order of gpio_set/spi_send_byte in mcp2515_instruction

**Pull request merged**